_posts/2025-08-07-Diving-into-services-with-OpenSearch-and-OpenTelemetry.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: post
+title: "Diving deep into distributed microservices with OpenSearch and OpenTelemetry"
+category: blog
+tags: [observability, traces, services, opentelemetry, opensearch-3-1]
+authors:
+    - pshenoy
+    - tackadam
+date: 2025-08-07
+categories:
+  - technical-posts
+meta_keywords: observability, traces, services, OpenTelemetry, OpenSearch 3.1, Investigation, 
+meta_description: Learn how to use OpenSearch 3.1 and OpenTelemetry for end-to-end observability, troubleshooting, and root cause analysis in distributed microservices. This blog post demonstrates a practical workflow for identifying, investigating, and resolving service issues using advanced tracing, service maps, and AI-powered features in OpenSearch Dashboards.
+---
+
+In modern distributed systems, understanding the interactions between microservices is crucial for identifying performance bottlenecks and diagnosing failures. In this blog post, we'll demonstrate the new features introduced in the Trace Analytics plugin in OpenSearch version 3.1, such as enhanced service map visualizations, advanced span grouping, and latency distribution charts. We'll also show how OpenTelemetry (OTel) Collector with OpenSearch Data Prepper can be used to instrument, collect, and ingest telemetry data into OpenSearch. Then, we'll walk through how to use OpenSearch Dashboards to explore, visualize, and analyze an observability investigation workflow. 
```

Comment:
Add link to OpenSearch 3.1: https://opensearch.org/blog/get-started-with-opensearch-3-1/